### PR TITLE
(VANAGON-136) Publish yaml settings during render

### PR DESCRIPTION
**Pull request by caseywilliams**
Writes a yaml representation of project settings to the workdir during
`render` (in addition to during `build`) when `publish_yaml_settings`
has been called in the project configuration.

We're hoping to use this to easily determine (without building everything completely) which settings have been updated for project/platform combinations in puppet-runtime PRs. This also removes the hard-coded `output` destination, which should have been how this method worked originally.

### CODE REVIEW

**Comment by underscorgan** [comment]
Can you split out the `workdir` changes into a separate commit?

File: lib/vanagon/driver.rb [hunk 1]
```diff
@@ -144,7 +144,7 @@ def run # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity
       @engine.ship_workdir(workdir)
       @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make} #{make_target})")
       @engine.retrieve_built_artifact(@project.artifacts_to_fetch, @project.no_packaging)
-      @project.publish_yaml_settings(@platform)
+      @project.publish_yaml_settings(workdir, @platform)
```
Comment (underscorgan) [inline lib/vanagon/driver.rb]: During the build is the workdir 'output' or a directory that gets synced back to output?

Comment (caseywilliams) [reply lib/vanagon/driver.rb]: Hmm, it looks like workdir is a temporary directory that's synced back to output, but now that I'm looking more at it, it seems like it'd be better to just use the `platform.output_dir` or something -- update coming soon.

**Comment by shrug** [comment]
@caseywilliams is this still on your radar?

**Comment by caseywilliams** [comment]
I doubt I'll have time to revisit this for a while -- I'll close it for now and reopen once I have something working and tested

**Pull request closed without merging**